### PR TITLE
doc: add rgw lc related config options

### PR DESCRIPTION
**Pull request by mikulely**
http://tracker.ceph.com/issues/19280

### CODE REVIEW

File: doc/radosgw/config-ref.rst [hunk 1]
```diff
@@ -1115,6 +1115,39 @@ Keystone Settings
 :Type: Boolean
 :Default: ``true``
 
+
+``rgw enable lc threads``
```
Comment (cbodley) [inline doc/radosgw/config-ref.rst]: could you add a section header for these so they're not confused with the `Keystone Settings` section above?

Comment (mikulely) [inline doc/radosgw/config-ref.rst]: added.

File: doc/radosgw/config-ref.rst [hunk 2]
```diff
@@ -1115,6 +1115,44 @@ Keystone Settings
 :Type: Boolean
 :Default: ``true``
 
+
+
+LifeCycle Management Settings
```
Comment (cbodley) [inline doc/radosgw/config-ref.rst]: can we use `Lifecycle` here to be consistent with [amazon docs](http://docs.aws.amazon.com/AmazonS3/latest/dev/object-lifecycle-mgmt.html)?

**Comment by cbodley** [comment]
could you please update your commit message to say `Fixes: http://tracker.ceph.com/issues/19280`

File: doc/radosgw/config-ref.rst (hunk 2 above)
Comment (mikulely) [inline doc/radosgw/config-ref.rst]: good point

**Comment by mikulely** [comment]
@cbodley  addressed. ;-)

File: doc/radosgw/config-ref.rst [hunk 3]
```diff
@@ -1115,6 +1115,44 @@ Keystone Settings
...
+:Description: The maximum number of objects that may be handled by
+              lifecycle management thread in one processing cycle.
+              Shoud not be larger than 7877(HASH_PRIME).
+:Type: Integer
+:Default: ``32``
+
+
+``rgw lifecycle work time``
+
+:Description: The period for lifecycle management thread can run. The
```
Comment (cbodley) [inline doc/radosgw/config-ref.rst]: this sentence is a little awkward. maybe something like this?
`The period of time that the lifecycle management thread is allowed to run each day.`

Comment (mikulely) [inline doc/radosgw/config-ref.rst]: ok. repushed.

File: doc/radosgw/config-ref.rst [hunk 4]
```diff
@@ -1115,6 +1115,45 @@ Keystone Settings
...
+:Description: The total run time in seconds for a single lifecycle processor.
+:Type: Integer
+:Default: 60
+
+
+``rgw lc max objs``
+
+:Description: The maximum number of objects that may be handled by
+              lifecycle management thread in one processing cycle.
+              Shoud not be larger than 7877(HASH_PRIME).
```
Comment (yehudasa) [inline doc/radosgw/config-ref.rst]: @mikulely the description here is incorrect. That's the max number of rados objects that hold the list of objects that are destined for trimming.

Comment (mikulely) [inline doc/radosgw/config-ref.rst]: @yehudasa  According to https://github.com/ceph/ceph/blob/master/src/rgw/rgw_lc.cc#L153 
maybe not only for trimming?

what about following:

 ```
The maximum number of rados objects(holding the list of 
buckets that enabled lifecycle management) that be handled by
 lifecycle management thread in one processing cycle.
```

**Pull request closed without merging**